### PR TITLE
Using Docker Stages to support no-argument docker bulding 

### LINKED ISSUE #6747: Make GATK docker buildable via a simple "docker build", with no required extra arguments
This change is necessary for the automatic building of GATK docker images in our large-scale testing framework.

### PR DESCRIPTION
**Pull request by jamesemery**
Running just "docker build ." in a gatk clone works with this branch. Unfortunately unless you then run a prune command you will end up with a 9 GB "waste" docker image from the first stage leftover. The build.docker script handles this situation by deleting the intermediate stage using tags...

Fixes #6747

### CODE REVIEW

**Comment by jamesemery** [comment]
Now to begin the process of making the travis tests work on this branch....

**Comment by fleharty** [comment]
Nifty, I learned about docker system prune...
Total reclaimed space: 9.254GB

File: .travis.yml [hunk 1]
```diff
@@ -158,15 +158,22 @@ script:
     fi;
     sudo docker images;
     echo ${TEST_TYPE};
-    sudo mkdir -p build/reports/;
-    sudo chmod -R a+w build/reports/;
+    sudo rm -r ./build;
```
Comment (lbergelson) [inline .travis.yml:161]: Why was this necessary?  Just precautionary?

**Comment by jamesemery** [comment]
@lbergelson this is finally working on travis! can you take a look and let me know your thoughts?

File: build_docker.sh [hunk 2]
```diff
@@ -112,22 +112,14 @@ if [ -n "${IS_PUSH}" ]; then
...
 
 echo "Building image to tag ${REPO_PRJ}:${GITHUB_TAG}..."
 if [ -n "${IS_PUSH}" ]; then
-    docker build -t ${REPO_PRJ}:${GITHUB_TAG} --squash --build-arg ZIPPATH=./unzippedJar .
+    docker build -t ${REPO_PRJ}:${GITHUB_TAG} --squash .
 else
-    docker build -t ${REPO_PRJ}:${GITHUB_TAG} --build-arg ZIPPATH=./unzippedJar .
+    docker build -t ${REPO_PRJ}:${GITHUB_TAG} .
 fi
+docker image prune -f --filter label=stage=gatkIntermediateBuildImage
```
Comment (lbergelson) [inline build_docker.sh:122]: Could you add a comment or echo here to say what this is for?

File: .travis.yml [hunk 3]
```diff
@@ -158,15 +158,22 @@ script:
...
+    ./gradlew clean shadowTestClassJar shadowTestJar;
+    mkdir ./testJars;
+    echo $( find ./build/libs/ -name "gatk*test.jar");
```
Comment (lbergelson) [inline .travis.yml:164]: Could you add some comments here explaining what this is for?

File: src/test/java/org/broadinstitute/hellbender/engine/PipelineSupportIntegrationTest.java [hunk 4]
```diff
@@ -16,8 +16,8 @@
     // Asserting that Picard programs are able to pipe their output without errors
     public void testPipeForPicardTools() {
         File output = createTempFile("testOutput",".bam");
-        BaseTest.runProcess(ProcessController.getThreadLocal(), new String[]{"/bin/sh","-c"," ./gatk SortSam -I "+INPUT_BAM+" -O /dev/stdout -SO coordinate " +
-                "| ./gatk SetNmMdAndUqTags -I /dev/stdin -O "+ output.getAbsolutePath()+ " --CREATE_INDEX true -R "+b37_reference_20_21});//.split(" "));
+        BaseTest.runProcess(ProcessController.getThreadLocal(), new String[]{"/bin/sh","-c"," ${GATK_LAUNCH_SCRIPT} SortSam -I "+INPUT_BAM+" -O /dev/stdout -SO coordinate " +
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/engine/PipelineSupportIntegrationTest.java:19]: We should be careful about this substitution.  Can you have it default to `./gatk` if it doesn't exist?

File: Dockerfile [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
 
-# Location of the unzipped gatk bundle files
-ARG ZIPPATH
+RUN ls .
+ADD . /gatk
+WORKDIR /gatk
+
+RUN export GRADLE_OPTS="-Xmx4048m -Dorg.gradle.daemon=false" && /gatk/gradlew clean collectBundleIntoDir shadowTestClassJar shadowTestJar -Drelease=$RELEASE
+RUN cp -r $( find /gatk/build -name "*bundle-files-collected" )/ /gatk/unzippedJar/
+RUN unzip -o -j $( find /gatk/unzippedJar -name "gatkPython*.zip" ) -d /gatk/unzippedJar/scripts
```
Comment (lbergelson) [inline Dockerfile:11]: comments about the test jar would be useful to explain that it's necessary plumbing for tests but not for the build itself

**Comment by jamesemery** [comment]
@lbergelson can you re-review? i have responded to your comments

**Comment by lbergelson** [comment]
@jamesemery Did you forget to push some changes?  I don't see any of the comments I asked for or the changes to what happens when the GATK_LAUNCH_SCRIPT is null.
